### ISSUE TITLE
plist parsing code needed

### ISSUE THREAD

**Issue by rtyler** [freebsd/openlaunchd#6]
Since `launchd` relies on CoreFoundation to parse its plist job configuration files, we'll need something to parse the plist files.

[CF-lite](http://opensource.apple.com/source/CF/CF-744.12/) exists, but is under the Apple Public Source License, so that is not desirable.

See `src/core.c` and `man/launchd.plist.5` for more documentation

**Comment by cneira** [comment]
gnustep has a property list parser, you could take a look at it

**Comment by netbsduser** [comment]
@cneira I presume that would be under the GPL, which is presumably to be avoided for something that hopes to become part of FreeBSD base some day.

**Comment by CameronNemo** [comment]
Is it not also Objective C only? Or can it be linked from plain C?

**Comment by CameronNemo** [comment]
Also, it could be used as a temporary solution until a non-GPL version can be created.

**Comment by cneira** [comment]
Yes, it could be used as a temporary solution, as far as I know objective-c is a superset of C, so the would be no issues.

**Comment by davidmurray** [comment]
Check out libplist. http://cgit.sukimashita.com/libplist.git

**Comment by cneira** [comment]
I would like to take care of this issue

**Comment by jbergstroem** [comment]
@Cykey libplist also is (L)GPL, making it unsuitable for the same reason.

**Comment by jbergstroem** [comment]
Wouldn't any XML parser work here (with some sugar on top for key/value lookups)?

**Comment by chris-turner137** [comment]
Hi Guys,
I'm currently working on a property list library for C, it's very much a work in progress and can be seen on my github account. I'm licensing it under the MIT license, which I presume is suitable. If you would like, when it's suitably ready I'll subtree merge it and rewrite the property list handling code in launchd to make use of it.
Chris

**Comment by rtyler** [comment]
@chris-turner137 I'm assuming you're referring to [libcplist](https://github.com/chris-turner137/libcplist)? Any help you have would be appreciated! I'm watching the repo now and look forward to seeing more of your work :)

**Comment by chris-turner137** [comment]
@rtyler Yes that's the one. What do you need for it to be considered ready? I'm implementing the plain text first and then XML (probably using expat). If I'm realistic, I'm not really motivated to implement support for the binary format.

**Comment by rtyler** [comment]
@chris-turner137 The XML-based support is most interesting for the purposes of openlaunchd, fwiw.
